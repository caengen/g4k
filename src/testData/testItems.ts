import { Item } from "../types";

export const testAItems = [
  {
    id: "1",
    name: "A",
    value: 50
  },
  {
    id: "1a",
    name: "A",
    value: 50
  },
  {
    id: "1b",
    name: "A",
    value: 50
  },
];

export const testBItems = [
  {
    id: "2",
    name: "B",
    value: 30
  },
  {
    id: "2a",
    name: "B",
    value: 30
  }
];

export const testCItems = [
  {
    id: "3",
    name: "C",
    value: 20
  },
  {
    id: "3a",
    name: "C",
    value: 20
  },
  {
    id: "3b",
    name: "C",
    value: 20
  },
];

export const testDItems = [
  {
    id: "4",
    name: "D",
    value: 15
  },
  {
    id: "4a",
    name: "D",
    value: 15
  },
  {
    id: "4b",
    name: "D",
    value: 15
  },
];

export const testItems = [
  ...testAItems,
  ...testBItems,
  ...testCItems,
  ...testDItems
];