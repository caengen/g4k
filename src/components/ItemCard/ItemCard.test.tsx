import React from "react";
import { ItemCard } from ".";
import { testItems } from "../../testData";
import { shallow } from "enzyme";

describe("ItemCard", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<ItemCard item={testItems[0]} />)
    expect(wrapper).toMatchSnapshot();
  })
});