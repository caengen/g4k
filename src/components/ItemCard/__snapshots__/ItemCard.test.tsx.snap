// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard renders correctly 1`] = `
<styled.div
  onClick={[Function]}
>
  A
</styled.div>
`;
